wandb/sdk/wandb_run.py

```diff
@@ -854,6 +854,18 @@ def _redirect(self, stdout_slave_fd, stderr_slave_fd):
         console = self._settings._console
         logger.info("redirect: %s", console)
 
+        if os.name == "nt" and sys.version_info >= (3, 6):
+            legacy_env_var = "PYTHONLEGACYWINDOWSSTDIO"
+            if legacy_env_var not in os.environ:
+                msg = (
+                    "Set %s environment variable to enable"
+                    " proper console logging on Windows. Falling "
+                    "back to monkey patching stdout/err." % legacy_env_var
+                )
+                print(msg)
+                logger.info(msg)
+                console = self._settings.Console.NOTEBOOK
```

Comment:
The decision of what console to use shouldnt be spread out among the code, it should in wandb_settings.py be at:
 @property
    def _console(self) -> SettingsConsole: